tests/components/nest/test_api.py

```diff
@@ -55,7 +59,7 @@ async def test_auth(
 
     async def async_new_subscriber(
         creds, subscription_name, event_loop, async_callback
-    ):
+    ) -> GoogleNestSubscriber | None:
```

Comment:
Try adding at the top of the module:

```py
from __future__ import annotations
```

Otherwise I'm not sure what's wrong.